Added Teams

vicky5124 opened the pull request:
Fixes #7

vicky5124 committed:
Moved Ratelimit identifier to it's own function
	modified:   src/main.rs

vicky5124 committed:
Initial teams implementation
	new file:   migrations/20210108093909_teams.sql
	modified:   sqlx-data.json
	modified:   src/main.rs
	modified:   src/model.rs
	modified:   src/routes/login.rs
	modified:   src/routes/mod.rs
	new file:   src/routes/teams.rs
	modified:   src/routes/upload.rs
	modified:   src/routes/users.rs
	modified:   src/routes/verification.rs

vicky5124 committed:
First try 😎
	modified:   src/routes/teams.rs

vicky5124 committed:
Added simple front end for creating a team
	new file:   static/create_table.html

vicky5124 committed:
Added token generation
	modified:   src/main.rs
	modified:   src/routes/login.rs
	modified:   src/routes/teams.rs

vicky5124 committed:
Updated actix-ratelimit to release
	modified:   Cargo.lock
	modified:   Cargo.toml

vicky5124 committed:
Added mod transfering to team
	modified:   src/main.rs
	modified:   src/routes/teams.rs

vicky5124 committed:
Stop forgetting preparation erreeeeeerererreeeee
	modified:   sqlx-data.json

Conqu3red commented:
everything looks good but what about endpoints to invite/add people to a team, or to view the members in a team?

vicky5124 committed:
Added team invites
	new file:   migrations/20210124180840_team_invites.sql
	new file:   migrations/20210124194502_deduplicate_team_members.sql
	modified:   src/main.rs
	modified:   src/routes/teams.rs
	new file:   src/utils/invite.rs
	modified:   src/utils/mod.rs

vicky5124 committed:
Update preparation
	modified:   sqlx-data.json

vicky5124 commented:
@Conqu3red Invites have been implemented.
knowing what teams you are on will be added on #24

Conqu3red reviewed (approved):
Everything looks good!

vicky5124 committed:
Merge branch 'main' into teams
